xds/src/main/java/io/grpc/xds/EnvoyProtoData.java

```diff
@@ -339,4 +339,270 @@ public String toString() {
           .toString();
     }
   }
+
+  static final class CidrRange {
```

Comment:
This is not SDS specific: only that SDS will be the first one to use it. This is server side Listener stuff that is going to be needed for more server side features. We can split client vs server. @voidzcy what do you think?

Reply:
Hmm... It is subtle. There is no clear distinction of which protos are used for client side and which are for server side. Some of them are used both client and server sides. Yes, I agree we probably do not want to dump everything here. Do we have a better solution? @creamsoup

Reply:
i like to have bunch of smaller classes. so i would split into xdsProtoData like EdsProtoData, CdsProtoData, etc. eventually something similar to envoy proto's package structure (more like flattened way). for this PR, possibly `SdsProtoData` will have SDS specific protos. this can help readers to understand which xds service it interacts with. 

i used this converter / data / proto separation in bazel, it works very well. but, not so well in other build system imo.

Reply:
@creamsoup can you clarify your comment? Preferring "bunch of smaller classes" is actually opposite of "flattened", so I am a bit confused. Also talking to @ejona86 not having smaller classes (on the xDS side) seems to be the preferred way. So on the xDS side we should create definitions that match our usage instead of just mirroring those on the Envoy proto side.

Reply:
oops yes.
flattened as package level. the data classes from various xds protos (has many sub-packages) can be in a single java package. but they can have their own classes such as Cds, Eds, Lds, Rds, Srds, Discovery, Common (common for listener, endpoint etc) etc. 

what do you mean not having smaller classes seems to be preferred? i meant the smaller as each DS (sorry if this was not clear), current code is somewhat in that structure (Cds, Eds, Sds, etc).  
i am referring envoy proto structure, because it is organized in the usage not having everything in one class. seems like you are interpreting it in too extreme case (all in one class vs as tiny as message?)

Reply:
@creamsoup okay, thanks for the clarification. So regarding not preferring smaller classes, let's consider an example: `Listener` contains (list of) `FilterChain` and a `FilterChain` contains `DownstreamTlsContext` in the original Envoy protos. We don't have to replicate this structure on our side but just define the top level `Listener` with all fields expanded "inline" unless the types `FilterChain` and `DownstreamTlsContext` are required elsewhere. Note this is just an example to illustrate the point.

On the Envoy side, they are not defined by us and so have less control over how types/protos are defined - but we don't have to follow the same structure.

Reply:
yes, that's why i specifically says "similar structure". if it is same, no reason to have data classes. also, this is why I suggested to rename the Listener to something else.
your response still sounds like you are thinking too micro level, my comments was for overall structure. I completely agree that Listener can have its own fields as long as it can represent current Listener structure. What I was saying is this `EnvoyProtoData` is too big. a single class represents the whole envoy protos sounds very extreme, so we may need to split into smaller chunks.

Reply:
Created a separate EnvoyServerProtoData class